cranelift/interpreter/src/interpreter.rs

```diff
@@ -965,4 +975,57 @@ mod tests {
 
         assert_eq!(result, vec![DataValue::F32(Ieee32::with_float(1.0))])
     }
+
+    #[test]
+    fn misaligned_store_traps() {
+        let code = "
+        function %test() {
+            ss0 = explicit_slot 8
```

Comment:
Can we change this to be `explicit_slot 16`? The reason is that this store is accessing memory out of bounds, and the interpreter didn't catch it!

Technically the [stack slots can be reordered](https://docs.rs/cranelift-codegen/latest/cranelift_codegen/ir/entities/struct.StackSlot.html), so we could reorder this slot to be at the end, and then we would access past the end of the stack. The interpreter never does that so it would never detect that case.

I've opened #5927 to track this!